### PR TITLE
[WIP] Add integration specs for not autoremembering flags

### PR DESCRIPTION
**Pull request by smlance**
@pducks32 and I thought that modifying `@use_current` in `Bundler.settings#[]=` was the most sensible way to test that flags are not autoremembered between commands, since we cannot modify `Bundler::VERSION` within specs.

Since users can run `bundle config use_current true`, should we document this setting, or should we leave it as a hidden setting used solely in the added specs?

Relevant: https://github.com/bundler/bundler/pull/3744

### CODE REVIEW

**Comment by segiddins** [comment]
@smlance yeah, you can modify the version in specs :P

File: lib/bundler/settings.rb [hunk 1]
```diff
@@ -33,7 +33,8 @@ def [](name)
     # remembered arguments.
     # See https://trello.com/c/yGsPNDpg
     def []=(key, value)
-      @use_current ||= (Bundler::VERSION.split(".")[0].to_i >= 2)
+      @use_current ||= Bundler.settings[:use_current] ||
+                       (Bundler::VERSION.split(".")[0].to_i >= 2)
```
Comment (segiddins) [inline lib/bundler/settings.rb]: since this is always on the `2.0` branch, this line will always return `true`

**Comment by smlance** [comment]
> yeah, you can modify the version in specs :P
How?

> since this is always on the 2.0 branch, this line will always return true
The problem is that the version number in 2-0-dev is currently set to 1.10.4.

**Comment by segiddins** [comment]
@smlance a PR before this that changes the version and fixes the failing specs because of that would be excellent!

You can use the `simulate_bundler_version` helper to change the bundler version inside the given block.

**Comment by smlance** [comment]
Thanks; I'll look into both of those shortly!

**Comment by smlance** [comment]
https://github.com/bundler/bundler/pull/3793

**Comment by segiddins** [comment]
@smlance what's the status of this?

**Comment by smlance** [comment]
@segiddins It's on hold until we implement not autoremembering flags.

**Comment by segiddins** [comment]
This'll need a rebase

**Pull request closed without merging**